File: AssignmentQ3.java
////////////////////////////////////////////////////////////////////////////////////////////////////////
//
// Function     : CheckPerfect
// Description  : check the Number is Perfect or Not
// Auther       : Sanyam BhupendraKumar Ravne
// Date         : 07/11/2025
//
////////////////////////////////////////////////////////////////////////////////////////////////////////

class Logic
{
    void CheckPerfect(int Num)
    {
        int iSum = 0;

        for(int i = 1; i < Num; i++)
        {
            if(Num % i == 0)
            iSum += i;
        }
        if(iSum == Num && Num > 0)
        {
            System.out.println( Num+ " is a Perfect Number");
        }
        else
        {
            System.out.println(Num+ " is Not a Perfect Number");
        }
    }
}
class AssignmentQ3
{
    public static void main(String A[])
    {
        Logic lobj = new Logic();
        lobj.CheckPerfect(6);
    }
}
////////////////////////////////////////////////////////////////////////////////////////////////////////
//
// Input : 6        Output : is a Perfect Number
//
////////////////////////////////////////////////////////////////////////////////////////////////////////

File: AssignmentQ3.cpp
#include<iostream>
using namespace std;

///////////////////////////////////////////////////////////////////////
//
// Function : Division
// Description : Division of two Number Generic
// Auther   : Sanyam BhupendraKumar Ravne
// Date : 10-01-2026
//
///////////////////////////////////////////////////////////////////////

template<class T>
T Division(T no1, T no2)
{
    T Div = 0;

    Div = no1 / no2;

    return Div;
}

int main()
{
    int iRet = Division(20,5);
    cout<<"Division of two Integer is : "<<iRet<<"\n";

    double dRet = Division(20.5644,5.1564);
    cout<<"Division of two Double is : "<<dRet<<"\n";

    return 0;
}
/*
Input : 20,5            Output : Division of two Integer is : 4
input : 20.5644,5.1564  Output : Division of two Double is : 3.98813
*/